### PR TITLE
add reporter for E2E

### PR DESCRIPTION
**Pull request by wackxu**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

add reporter for E2E, so we can get the detail report when e2e run finish

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Review by chendave — COMMENTED** [review]
NOTE that we on the process of Ginkgo v2 migration, this is WIP both in K8S and kubeedge, while custom reporter is not supported by Ginkgo V2.

I am not saying this is not useful, but something that might be hold until Ginkgo V2 is landing in this project.

**Comment by fisherxu** [comment]
What is the `Ginkgo v2 migration` working progress and release plan in kubernetes? @chendave 
cc @vincentgoat @Congrool

**Comment by chendave** [comment]
@fisherxu , it is currently pending on the sig-testing, no commitment on the timeline, but I believe that will happen in 1.25 release cycle.

File: tests/e2e/deployment/e2e_test.go [hunk 1]
```diff
@@ -59,5 +64,24 @@ func TestE2E(t *testing.T) {
...
 	})
 
-	RunSpecs(t, "kubeedge App Deploymet Suite")
+	// Run tests through the Ginkgo runner with output to console + JUnit for Jenkins
+	var r []Reporter
+	if utils.LoadConfig().ReportDir != "" {
+		if err := os.MkdirAll(utils.LoadConfig().ReportDir, 0755); err != nil {
+			klog.Errorf("Failed creating report directory: %v", err)
+		} else {
+			r = append(r, reporters.NewJUnitReporter(path.Join(utils.LoadConfig().ReportDir, fmt.Sprintf("junit_%v.xml", utils.LoadConfig().ReportPrefix))))
```
Comment (chendave) [inline tests/e2e/deployment/e2e_test.go:73]: Ginkgo V2 doesn't need to explicitly implement a reporter for junit.  If this pr wants to only implement junit reporter,  then we shouldn't merge this for Ginkgo V2.

**Comment by chendave** [comment]
since Ginkgo v2 doesn't need the junit report, and we are on v2 right now, so can close this one.

thanks for your contribution!

**Pull request closed without merging**